akka-docs/src/main/paradox/typed/cluster-membership.md

```diff
@@ -58,60 +58,74 @@ Note that the node might already have been shutdown when this event is published
 of at least one other node.
  * `ClusterEvent.ReachableMember` - A member is considered as reachable again, after having been unreachable.
 All nodes that previously detected it as unreachable has detected it as reachable again.
- 
+
 ## Membership Lifecycle
 
-A node begins in the `joining` state. Once all nodes have seen that the new
-node is joining (through gossip convergence) the `leader` will set the member
-state to `up`.
+A node is introduced to the cluster by invoking the `join` action which puts the node in the `joining` state.
+Once all nodes have seen that the new node is joining (through @ref[gossip convergence](cluster-concepts.md#gossip-convergence)) the
+@ref[`leader`](cluster-concepts.md#leader) will set the member state to `up`.
 
 If a node is leaving the cluster in a safe, expected manner, for example through @ref[coordinated shutdown](../coordinated-shutdown.md), 
-it switches to the `leaving` state. Once the leader sees the convergence on the node in the
+it invokes the `leave` action which switches it to the `leaving` state. Once the leader sees the convergence on the node in the
 `leaving` state, the leader will then move it to `exiting`.  Once all nodes
 have seen the exiting state (convergence) the `leader` will remove the node
 from the cluster, marking it as `removed`.
 
 If a node is `unreachable` then gossip convergence is not possible and therefore
-any `leader` actions are also not possible (for instance, allowing a node to
-become a part of the cluster). To be able to move forward the state of the
-`unreachable` nodes must be changed. It must become `reachable` again or marked
-as `down`. If the node is to join the cluster again the actor system must be
-restarted and go through the joining process again. If new incarnation of the unreachable
-node tries to rejoin the cluster old incarnation will be marked as `down` and new
-incarnation can rejoin the cluster without manual intervention. 
+most `leader` actions are impossible (for instance, allowing a node to
```

Comment:
I added a bit more explanations here about the reasoning about why things are the way they are.